platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslCompilerPlugins.kt

```diff
@@ -68,6 +69,9 @@ abstract class KotlinDslCompilerPlugins : Plugin<Project> {
                     "-XXLanguage:+DisableCompatibilityModeForNewInference",
                     "-XXLanguage:-TypeEnhancementImprovementsInStrictMode",
                 )))
+                // Set this back to a warning for now, as this plugin is frequently used without toolchains specifying a JVM target, and it causes errors when using newer JDKs.
+                // This can be removed when https://youtrack.jetbrains.com/issue/KT-66919 is fixed.
+                kotlinCompile.jvmTargetValidationMode.set(JvmTargetValidationMode.WARNING)
```

Comment:
💭 Please create an issue on our side to track the resolution of the upstream issue so we don't forget to remove this when applicable.